ballerina/root_field_visitor.bal

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import graphql.parser;
+
+class RootFieldVisitor {
+    *parser:Visitor;
+
+    private parser:FieldNode[] fieldNodes;
```

Comment:
If the `fieldNode` variable is nullable, an error handling is required to assure it is not not null before it has been used in another function. Since the `operationNode` is already validated there is no need of error handling required here. I used an array here, because it can be used to return non nullable `fieldNode`.

Reply:
No, that's not the way to handle it.

A nullable variable does not need error handling. It only needs to check whether the value is null or not. In other words, if we have a value or not. Using an array to bypass that check does not solve the original issue. What if the array is an empty one? The same issue occurs.

Even from the memory point of view, this has a negative impact.

Reply:
BTW can `fieldNode` be nullable at this point? Doesn't it validate at the `subscription_visitor`?

Reply:
It should not be null. But the point is, that we cannot initialize it to the exact type in the beginning. So we have to use a nullable type. Otherwise, an option is not to define a class variable but use a local variable. But I don't think that would work in this case.